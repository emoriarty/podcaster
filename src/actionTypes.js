export const FETCH_COUNTRIES = 'itunes/FETCH_COUNTRIES';
export const FETCH_COUNTRIES_SUCCESS = 'itunes/FETCH_COUNTRIES_SUCCESS';
export const FETCH_COUNTRIES_FAIL = 'itunes/FETCH_COUNTRIES_FAIL';

export const FETCH_MEDIA_TYPES = 'itunes/FETCH_MEDIA_TYPES';
export const FETCH_MEDIA_TYPES_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_SUCCESS';
export const FETCH_MEDIA_TYPES_FAIL = 'itunes/FETCH_MEDIA_TYPES_FAIL';

export const FETCH_COMMON_TRANSLATIONS = 'itunes/FETCH_COMMON_TRANSLATIONS';
export const FETCH_COMMON_TRANSLATIONS_SUCCESS = 'itunes/FETCH_COMMON_TRANSLATIONS_SUCCESS';
export const FETCH_COMMON_TRANSLATIONS_FAIL = 'itunes/FETCH_COMMON_TRANSLATIONS_FAIL';

export const FETCH_MEDIA_TYPES_TRANSLATIONS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL';
