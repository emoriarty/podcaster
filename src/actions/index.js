export { default as fetchCountries } from './itunes/fetchCountries'
export { default as fetchMediaTypes } from './itunes/fetchMediaTypes'
export { default as fetchCommonTranslations } from './itunes/fetchCommonTranslations'
export { default as fetchMediaTypesTranslations } from './itunes/fetchMediaTypesTranslations'
export { default as fetchPodcasts } from './itunes/fetchPodcasts'
export { default as fetchTranslations } from './itunes/fetchTranslations'
