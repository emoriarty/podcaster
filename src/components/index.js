export * from './Expansion'
export * from './Layouts'
export * from './ListItems'
export * from './Texts'
