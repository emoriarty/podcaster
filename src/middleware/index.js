export { default as yqlMiddleware } from './yql'
export { default as rssMiddleware } from './rss'
