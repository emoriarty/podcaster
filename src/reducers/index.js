import { combineReducers } from 'redux'
import countries from './countries'
import entities from './entities'
import mediaTypes from './mediaTypes'
import settings from './settings'
import translations from './translations'
import podcasts from './podcasts'

const app = combineReducers({
  entities,
  countries,
  mediaTypes,
  settings,
  translations,
  podcasts
})

export default app
