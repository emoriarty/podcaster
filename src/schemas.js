import { schema } from 'normalizr'

export const country = new schema.Entity(
  'countries',
  {},
  { idAttribute: 'country_code' }
)

export const mediaType = new schema.Entity('mediaTypes')

export const podcast = new schema.Entity(
  'podcasts',
  {},
  {
    idAttribute: (value, parent, key) => {
      return value.id.attributes['im:id']
    }
  })

export const podcastFeed = { feed: { entry: new schema.Array(podcast) } }
