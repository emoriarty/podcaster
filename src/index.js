import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import appReducer from './reducers';
import { fetchCountries } from './actions';

const loggerMiddleware = createLogger();

window.addEventListener('load', () => {
  // jsonp(
  //   'https://itunes.apple.com/search?term=carne&media=podcast',
  //   (err, response) => console.log(err, response)
  // );

  let store = createStore(
    appReducer,
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware
    )
  );

  console.log('initialState', store.getState());

  const unsubscribe = store.subscribe(() => console.log(store.getState()));

  store
    .dispatch(fetchCountries())
    .then(data => {
      console.log('success', data);
    })
    .catch(reason => {
      console.error(reason);
    });

  unsubscribe();
});
