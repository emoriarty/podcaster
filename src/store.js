import { createStore, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { yqlMiddleware } from './middleware'
import appReducer from './reducers'

const loggerMiddleware = createLogger()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default () =>
  createStore(
    appReducer,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        yqlMiddleware,
        loggerMiddleware
      )
    )
  )
