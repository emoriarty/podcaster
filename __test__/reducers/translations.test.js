import reducer from '../../src/reducers/translations'
import * as types from '../../src/actionTypes'

const responses = {
  [types.FETCH_COMMON_TRANSLATIONS_SUCCESS]:
    readJson('translations-common'),
  [types.FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS]: 
    readJson('translations-media-types')
}

const startActions = [
  types.FETCH_COMMON_TRANSLATIONS,
  types.FETCH_MEDIA_TYPES_TRANSLATIONS
]
const successActions = [
  types.FETCH_COMMON_TRANSLATIONS_SUCCESS,
  types.FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS
]
const failActions = [
  types.FETCH_COMMON_TRANSLATIONS_FAIL,
  types.FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL
]

function isFetching (store, type) {
  switch (type) {
    case types.FETCH_COMMON_TRANSLATIONS:
    case types.FETCH_COMMON_TRANSLATIONS_SUCCESS:
    case types.FETCH_COMMON_TRANSLATIONS_FAIL:
      return store.isFetchingCommon
    case types.FETCH_MEDIA_TYPES_TRANSLATIONS:
    case types.FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS:
    case types.FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL:
      return store.isFetchingMediaType
  }
}

let store = {}

describe('translations reducer', () => {
  it('should return the initial state', () => {
    store = reducer(undefined, {})
    expect(store).toMatchSnapshot()
  })

  startActions.forEach((type) =>
    it(`when action ${type} sets isFecthing property to true`, () => {
      const action = { type }
      store = reducer(store, action)

      expect(isFetching(store, type)).toBeTruthy()
    })
  )

  successActions.forEach((type, index, list) =>
    it(`should handle ${type}`, () => {
      const action = {
        type,
        payload: responses[type]
      }
      store = reducer(store, action)

      expect(store).toMatchSnapshot()
      expect(isFetching(store, type)).toBeFalsy()
    })
  )

  failActions.forEach((type) =>
    it(`should handle ${type}`, () => {
      const action = {
        type,
        error: new Error()
      }

      expect(reducer({}, action)).toMatchSnapshot()
      expect(isFetching(store, type)).toBeFalsy()
    })
  )
})
