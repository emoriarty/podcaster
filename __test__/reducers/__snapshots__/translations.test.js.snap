// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translations reducer should handle itunes/FETCH_COMMON_TRANSLATIONS_FAIL 1`] = `
Object {
  "data": Object {},
  "error": [Error],
  "isFetchingCommon": false,
  "isFetchingMediaType": false,
}
`;

exports[`translations reducer should handle itunes/FETCH_COMMON_TRANSLATIONS_SUCCESS 1`] = `
Object {
  "data": Object {
    "Country": "País",
    "Feed_Type": "Tipo de feed",
    "Genre": "Género",
    "Media_Type": "Tipo de contenido",
    "No": "No",
    "Size": "Tamaño",
    "Yes": "Sí",
    "feed_country": Object {
      "ae": "EAU",
      "ag": "Antigua y Barbuda",
      "ai": "Anguila",
      "al": "Albania",
      "am": "Armenia",
      "ao": "Angola",
      "ar": "Argentina",
      "ar_en": "Argentina (English)",
      "at": "Austria",
      "at_en": "Austria (English)",
      "au": "Australia",
      "az": "Azerbaiyán",
      "bb": "Barbados",
      "be": "Bélgica",
      "be_en": "Belgium (English)",
      "be_fr": "Belgium (French)",
      "bf": "Burkina Faso",
      "bg": "Bulgaria",
      "bh": "Bahréin",
      "bj": "Benín",
      "bm": "Bermudas",
      "bn": "Brunéi",
      "bo": "Bolivia",
      "bo_en": "Bolivia (English)",
      "br": "Brasil",
      "br_en": "Brazil (English)",
      "bs": "Bahamas",
      "bt": "Bután",
      "bw": "Botsuana",
      "by": "Bielorrusia",
      "bz": "Belice",
      "bz_es": "Belize (Spanish)",
      "ca": "Canadá",
      "ca_fr": "Canada (French)",
      "cg": "República del Congo",
      "ch": "Suiza",
      "ch_en": "Switzerland (English)",
      "ch_fr": "Switzerland (French)",
      "ch_it": "Switzerland (Italian)",
      "cl": "Chile",
      "cl_en": "Chile (English)",
      "cn": "China",
      "cn_en": "China (English)",
      "co": "Colombia",
      "co_en": "Colombia (English)",
      "cr": "Costa Rica",
      "cr_en": "Costa Rica (English)",
      "cv": "Cabo Verde",
      "cy": "Chipre",
      "cz": "República Checa",
      "de": "Alemania",
      "de_en": "Germany (English)",
      "dk": "Dinamarca",
      "dk_en": "Denmark (English)",
      "dm": "Dominica",
      "dm_en": "Dominica (English)",
      "do": "República Dominicana",
      "do_en": "Dominican Republic (English)",
      "dz": "Argelia",
      "ec": "Ecuador",
      "ec_en": "Ecuador (English)",
      "ee": "Estonia",
      "eg": "Egipto",
      "es": "España",
      "es_en": "Spain (English)",
      "fi": "Finlandia",
      "fi_en": "Finland (English)",
      "fj": "Fiyi",
      "fm": "Estados Federados de Micronesia",
      "fr": "Francia",
      "fr_en": "France (English)",
      "gb": "Reino Unido",
      "gd": "Granada",
      "gh": "Ghana",
      "gm": "Gambia",
      "gr": "Grecia",
      "gt": "Guatemala",
      "gt_en": "Guatemala (English)",
      "gw": "Guinea-Bisáu",
      "gy": "Guyana",
      "hk": "Hong Kong",
      "hk_en": "Hong Kong (English)",
      "hn": "Honduras",
      "hn_en": "Honduras (English)",
      "hr": "Croacia",
      "hu": "Hungría",
      "id": "Indonesia",
      "id_en": "Indonesia (English)",
      "ie": "Irlanda",
      "il": "Israel",
      "in": "India",
      "is": "Islandia",
      "it": "Italia",
      "it_en": "Italy (English)",
      "jm": "Jamaica",
      "jo": "Jordania",
      "jp": "Japón",
      "jp_en": "Japan (English)",
      "ke": "Kenia",
      "kg": "Kirguistán",
      "kh": "Camboya",
      "kn": "San Cristóbal y Nieves",
      "kr": "Corea del Sur",
      "kr_en": "Korea, Republic Of (English)",
      "kw": "Kuwait",
      "ky": "Islas Caimán",
      "kz": "Kazajistán",
      "la": "República Democrática Popular Lao",
      "lb": "Líbano",
      "lc": "Santa Lucía",
      "lk": "Sri Lanka",
      "lr": "Liberia",
      "lt": "Lituania",
      "lu": "Luxemburgo",
      "lu_en": "Luxembourg (English)",
      "lu_fr": "Luxembourg (French)",
      "lv": "Letonia",
      "md": "Moldavia",
      "mg": "Madagascar",
      "mk": "Macedonia",
      "ml": "Malí",
      "mn": "Mongolia",
      "mo": "Macao",
      "mo_en": "Macau (English)",
      "mr": "Mauritania",
      "ms": "Montserrat",
      "mt": "Malta",
      "mu": "Mauricio",
      "mw": "Malaui",
      "mx": "México",
      "mx_en": "Mexico (English)",
      "my": "Malasia",
      "my_en": "Malaysia (English)",
      "mz": "Mozambique",
      "na": "Namibia",
      "ne": "Niger",
      "ng": "Nigeria",
      "ni": "Nicaragua",
      "ni_en": "Nicaragua (English)",
      "nl": "Países Bajos",
      "nl_en": "Netherlands (English)",
      "no": "Noruega",
      "no_en": "Norway (English)",
      "np": "Nepal",
      "nz": "Nueva Zelanda",
      "om": "Omán",
      "pa": "Panamá",
      "pa_en": "Panama (English)",
      "pe": "Perú",
      "pe_en": "Peru (English)",
      "pg": "Papúa-Nueva Guinea",
      "ph": "Filipinas",
      "pk": "Pakistán",
      "pl": "Polonia",
      "pt": "Portugal",
      "pt_en": "Portugal (English)",
      "pw": "Palaos",
      "py": "Paraguay",
      "py_en": "Paraguay (English)",
      "qa": "Catar",
      "ro": "Rumanía",
      "ru": "Rusia",
      "ru_en": "Russia (English)",
      "sa": "Arabia Saudita",
      "sb": "Islas Salomón",
      "sc": "Seychelles",
      "se": "Suecia",
      "se_en": "Sweden (English)",
      "sg": "Singapur",
      "sg_en": "Singapore (English)",
      "si": "Eslovenia",
      "sk": "Eslovaquia",
      "sl": "Sierra Leona",
      "sn": "Senegal",
      "sr": "Surinam",
      "sr_en": "Suriname (English)",
      "st": "Santo Tomé y Príncipe",
      "sv": "El Salvador",
      "sv_en": "El Salvador (English)",
      "sz": "Suazilandia",
      "tc": "Turks and Caicos",
      "td": "Chad",
      "th": "Tailandia",
      "th_en": "Thailand (English)",
      "tj": "Tayikistán",
      "tm": "Turkmenistán",
      "tn": "Túnez",
      "tr": "Turquía",
      "tr_en": "Turkey (English)",
      "tt": "Trinidad y Tobago",
      "tw": "Taiwán",
      "tw_en": "Taiwan (English)",
      "tz": "Tanzania",
      "ua": "Ucrania",
      "ug": "Uganda",
      "us": "EE. UU.",
      "us_es": "United States (Spanish)",
      "uy": "Uruguay",
      "uy_en": "Uruguay (English)",
      "uz": "Uzbekistán",
      "vc": "St. Vincent and The Grenadines",
      "ve": "Venezuela",
      "ve_en": "Venezuela (English)",
      "vg": "Vírgenes Británicas, Islas",
      "vn": "Vietnam",
      "vn_en": "Vietnam (English)",
      "ye": "Yemen",
      "za": "Suráfrica",
      "zw": "Zimbabue",
    },
  },
  "error": null,
  "isFetchingCommon": false,
  "isFetchingMediaType": true,
}
`;

exports[`translations reducer should handle itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL 1`] = `
Object {
  "data": Object {},
  "error": [Error],
  "isFetchingCommon": false,
  "isFetchingMediaType": false,
}
`;

exports[`translations reducer should handle itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS 1`] = `
Object {
  "data": Object {
    "Country": "País",
    "Explicit_Content": "Contenido explícito",
    "Feed_Type": "Tipo de feed",
    "Genre": "Género",
    "Media_Type": "Tipo de contenido",
    "No": "No",
    "Size": "Tamaño",
    "Yes": "Sí",
    "action&adventure": "Acción y aventura",
    "african": "Cine africano",
    "alerttones": "Tonos de alerta",
    "alternative": "Alternativa",
    "animation": "Animación",
    "anime": "Anime",
    "appstore": "Apps para iOS",
    "art&architecture": "Arte y arquitectura",
    "arts": "Arte",
    "arts&entertainment": "Arte y espectáculos",
    "audiobooks": "Audiolibros",
    "audiobookslatino": "Audiolibros en español",
    "bigband": "Big band",
    "biographies&memoirs": "Biografías y memorias",
    "biography&memoir": "Biografías y autobiografías",
    "blues": "Blues",
    "bollywood": "Bollywood",
    "books": "Libros",
    "brazilian": "Música brasileña",
    "business": "Economía y empresa",
    "business&personalfinance": "Finanzas para empresas y particulares",
    "catalogs": "Catálogos",
    "children&teens": "Niños y adolescentes",
    "children'smusic": "Música para niños",
    "chinese": "Música china",
    "christian&gospel": "Inspiración",
    "classic": "Clásico",
    "classical": "Clásica",
    "classics": "Clásicos",
    "comedy": "Comedia",
    "comics&graphicnovels": "Cómics y novelas gráficas",
    "communications&media": "Medios de comunicación",
    "computers&internet": "Informática e Internet",
    "concertfilms": "Conciertos",
    "cookbooks,food&wine": "Cocina, gastronomía y vinos",
    "country": "Country",
    "dance": "Dance",
    "developertools": "Para desarrolladores",
    "disney": "Disney",
    "documentary": "Documentales",
    "drama": "Drama",
    "drama&poetry": "Drama y poesía",
    "easylistening": "Easy listening",
    "education": "Educación",
    "electronic": "Electrónica",
    "engineering": "Ingeniería",
    "enka": "Enka",
    "entertainment": "Entretenimiento",
    "featuredalbums": "Álbumes destacados",
    "feed_country": Object {
      "ae": "EAU",
      "ag": "Antigua y Barbuda",
      "ai": "Anguila",
      "al": "Albania",
      "am": "Armenia",
      "ao": "Angola",
      "ar": "Argentina",
      "ar_en": "Argentina (English)",
      "at": "Austria",
      "at_en": "Austria (English)",
      "au": "Australia",
      "az": "Azerbaiyán",
      "bb": "Barbados",
      "be": "Bélgica",
      "be_en": "Belgium (English)",
      "be_fr": "Belgium (French)",
      "bf": "Burkina Faso",
      "bg": "Bulgaria",
      "bh": "Bahréin",
      "bj": "Benín",
      "bm": "Bermudas",
      "bn": "Brunéi",
      "bo": "Bolivia",
      "bo_en": "Bolivia (English)",
      "br": "Brasil",
      "br_en": "Brazil (English)",
      "bs": "Bahamas",
      "bt": "Bután",
      "bw": "Botsuana",
      "by": "Bielorrusia",
      "bz": "Belice",
      "bz_es": "Belize (Spanish)",
      "ca": "Canadá",
      "ca_fr": "Canada (French)",
      "cg": "República del Congo",
      "ch": "Suiza",
      "ch_en": "Switzerland (English)",
      "ch_fr": "Switzerland (French)",
      "ch_it": "Switzerland (Italian)",
      "cl": "Chile",
      "cl_en": "Chile (English)",
      "cn": "China",
      "cn_en": "China (English)",
      "co": "Colombia",
      "co_en": "Colombia (English)",
      "cr": "Costa Rica",
      "cr_en": "Costa Rica (English)",
      "cv": "Cabo Verde",
      "cy": "Chipre",
      "cz": "República Checa",
      "de": "Alemania",
      "de_en": "Germany (English)",
      "dk": "Dinamarca",
      "dk_en": "Denmark (English)",
      "dm": "Dominica",
      "dm_en": "Dominica (English)",
      "do": "República Dominicana",
      "do_en": "Dominican Republic (English)",
      "dz": "Argelia",
      "ec": "Ecuador",
      "ec_en": "Ecuador (English)",
      "ee": "Estonia",
      "eg": "Egipto",
      "es": "España",
      "es_en": "Spain (English)",
      "fi": "Finlandia",
      "fi_en": "Finland (English)",
      "fj": "Fiyi",
      "fm": "Estados Federados de Micronesia",
      "fr": "Francia",
      "fr_en": "France (English)",
      "gb": "Reino Unido",
      "gd": "Granada",
      "gh": "Ghana",
      "gm": "Gambia",
      "gr": "Grecia",
      "gt": "Guatemala",
      "gt_en": "Guatemala (English)",
      "gw": "Guinea-Bisáu",
      "gy": "Guyana",
      "hk": "Hong Kong",
      "hk_en": "Hong Kong (English)",
      "hn": "Honduras",
      "hn_en": "Honduras (English)",
      "hr": "Croacia",
      "hu": "Hungría",
      "id": "Indonesia",
      "id_en": "Indonesia (English)",
      "ie": "Irlanda",
      "il": "Israel",
      "in": "India",
      "is": "Islandia",
      "it": "Italia",
      "it_en": "Italy (English)",
      "jm": "Jamaica",
      "jo": "Jordania",
      "jp": "Japón",
      "jp_en": "Japan (English)",
      "ke": "Kenia",
      "kg": "Kirguistán",
      "kh": "Camboya",
      "kn": "San Cristóbal y Nieves",
      "kr": "Corea del Sur",
      "kr_en": "Korea, Republic Of (English)",
      "kw": "Kuwait",
      "ky": "Islas Caimán",
      "kz": "Kazajistán",
      "la": "República Democrática Popular Lao",
      "lb": "Líbano",
      "lc": "Santa Lucía",
      "lk": "Sri Lanka",
      "lr": "Liberia",
      "lt": "Lituania",
      "lu": "Luxemburgo",
      "lu_en": "Luxembourg (English)",
      "lu_fr": "Luxembourg (French)",
      "lv": "Letonia",
      "md": "Moldavia",
      "mg": "Madagascar",
      "mk": "Macedonia",
      "ml": "Malí",
      "mn": "Mongolia",
      "mo": "Macao",
      "mo_en": "Macau (English)",
      "mr": "Mauritania",
      "ms": "Montserrat",
      "mt": "Malta",
      "mu": "Mauricio",
      "mw": "Malaui",
      "mx": "México",
      "mx_en": "Mexico (English)",
      "my": "Malasia",
      "my_en": "Malaysia (English)",
      "mz": "Mozambique",
      "na": "Namibia",
      "ne": "Niger",
      "ng": "Nigeria",
      "ni": "Nicaragua",
      "ni_en": "Nicaragua (English)",
      "nl": "Países Bajos",
      "nl_en": "Netherlands (English)",
      "no": "Noruega",
      "no_en": "Norway (English)",
      "np": "Nepal",
      "nz": "Nueva Zelanda",
      "om": "Omán",
      "pa": "Panamá",
      "pa_en": "Panama (English)",
      "pe": "Perú",
      "pe_en": "Peru (English)",
      "pg": "Papúa-Nueva Guinea",
      "ph": "Filipinas",
      "pk": "Pakistán",
      "pl": "Polonia",
      "pt": "Portugal",
      "pt_en": "Portugal (English)",
      "pw": "Palaos",
      "py": "Paraguay",
      "py_en": "Paraguay (English)",
      "qa": "Catar",
      "ro": "Rumanía",
      "ru": "Rusia",
      "ru_en": "Russia (English)",
      "sa": "Arabia Saudita",
      "sb": "Islas Salomón",
      "sc": "Seychelles",
      "se": "Suecia",
      "se_en": "Sweden (English)",
      "sg": "Singapur",
      "sg_en": "Singapore (English)",
      "si": "Eslovenia",
      "sk": "Eslovaquia",
      "sl": "Sierra Leona",
      "sn": "Senegal",
      "sr": "Surinam",
      "sr_en": "Suriname (English)",
      "st": "Santo Tomé y Príncipe",
      "sv": "El Salvador",
      "sv_en": "El Salvador (English)",
      "sz": "Suazilandia",
      "tc": "Turks and Caicos",
      "td": "Chad",
      "th": "Tailandia",
      "th_en": "Thailand (English)",
      "tj": "Tayikistán",
      "tm": "Turkmenistán",
      "tn": "Túnez",
      "tr": "Turquía",
      "tr_en": "Turkey (English)",
      "tt": "Trinidad y Tobago",
      "tw": "Taiwán",
      "tw_en": "Taiwan (English)",
      "tz": "Tanzania",
      "ua": "Ucrania",
      "ug": "Uganda",
      "us": "EE. UU.",
      "us_es": "United States (Spanish)",
      "uy": "Uruguay",
      "uy_en": "Uruguay (English)",
      "uz": "Uzbekistán",
      "vc": "St. Vincent and The Grenadines",
      "ve": "Venezuela",
      "ve_en": "Venezuela (English)",
      "vg": "Vírgenes Británicas, Islas",
      "vn": "Vietnam",
      "vn_en": "Vietnam (English)",
      "ye": "Yemen",
      "za": "Suráfrica",
      "zw": "Zimbabue",
    },
    "fiction": "Ficción",
    "fiction&literature": "Ficción y literatura",
    "finance": "Finanzas",
    "fitness&workout": "Fitness y gimnasia",
    "food&drink": "Comida y bebidas",
    "foreign": "Cine extranjero",
    "frenchpop": "Pop francés",
    "games": "Juegos",
    "games&hobbies": "Juegos y pasatiempos",
    "germanfolk": "Folk alemán",
    "germanpop": "Pop alemán",
    "government&organizations": "Gobierno y organizaciones",
    "graphics&design": "Diseño gráfico",
    "health": "Salud",
    "health&fitness": "Salud y forma física",
    "health&medicine": "Salud y medicina",
    "health,mind&body": "Salud y bienestar",
    "hip-hop/rap": "Hip-Hop/Rap",
    "history": "Historia",
    "holiday": "Música navideña",
    "horror": "Terror",
    "humor": "Humor",
    "independent": "Cine independiente",
    "indian": "Música de la India",
    "instrumental": "Instrumental",
    "itunesu": "iTunes U",
    "j-pop": "J-Pop ",
    "japanesecinema": "Cine japonés",
    "jazz": "Jazz",
    "jidaigeki": "Jidaigeki",
    "justadded": "Recién añadido",
    "k-pop": "K-Pop",
    "karaoke": "Karaoke",
    "kayokyoku": "Kayokyoku",
    "kids": "Niños",
    "kids&family": "Para toda la familia",
    "kids&youngadults": "Niños y jóvenes",
    "korean": "Música coreana",
    "koreancinema": "Cine correano",
    "language": "Idiomas",
    "languages": "Idiomas",
    "languagestudies": "Estudios lingüísticos",
    "latin": "Música latina",
    "latino": "Música latina",
    "latinotv": "TV latina",
    "law&politics": "Derecho y política",
    "lifestyle": "Estilo de vida",
    "lifestyle&home": "Estilo de vida y hogar",
    "literature": "Literatura",
    "macappstore": "Apps para Mac",
    "madefortv": "Hecho para TV",
    "mathematics": "Matemáticas",
    "medical": "Medicina",
    "medicine": "Medicina",
    "middleeastern": "Oriente Medio",
    "movies": "Películas",
    "music": "Música",
    "musicals": "Musicales",
    "musicdocumentaries": "Documentales de música",
    "musicfeaturefilms": "Películas musicales",
    "musicvideos": "Vídeos musicales",
    "mysteries&thrillers": "Misterio y suspense",
    "mystery": "Misterio",
    "navigation": "Navegación",
    "newage": "New age",
    "newapplications": "Apps nuevas",
    "newfreeapplications": "Apps gratuitas nuevas",
    "newpaidapplications": "Apps de pago nuevas",
    "newreleases": "Novedades",
    "news": "Noticias",
    "news&politics": "Noticias y política",
    "newsstand": "Quiosco",
    "nonfiction": "No ficción",
    "opera": "Ópera",
    "parenting": "Padres",
    "philosophy": "Filosofía",
    "photo&video": "Foto y vídeo",
    "photography": "Fotografía",
    "podcasts": "Podcasts",
    "politics&currentevents": "Política y actualidad",
    "pop": "Pop",
    "productivity": "Productividad",
    "professional&technical": "Profesional y técnico",
    "programs&performances": "Programas y actuaciones",
    "psychology&socialscience": "Psicología y ciencias sociales",
    "r&b/soul": "R&B/Soul",
    "realitytv": "Reality",
    "reference": "Referencia",
    "reggae": "Reggae",
    "regionalindian": "Regionales de la India",
    "religion&spirituality": "Religión y espiritualidad",
    "ringtones": "Tonos de llamada",
    "rock": "Rock",
    "romance": "Novela romántica",
    "russian": "Cine ruso",
    "sci-fi&fantasy": "Ciencia ficción y fantasía",
    "science": "Ciencias",
    "science&medicine": "Ciencia y medicina",
    "science&nature": "Ciencia y naturaleza",
    "scifi&fantasy": "Ciencia ficción y fantasía",
    "selfdevelopment": "Desarrollo personal",
    "shortfilms": "Cortometrajes",
    "singer/songwriter": "Cantautores",
    "socialnetworking": "Redes sociales",
    "socialscience": "Ciencias sociales",
    "society": "Sociedad",
    "society&culture": "Sociedad y cultura",
    "soundtrack": "Bandas sonoras",
    "speakers&storytellers": "Oradores y narradores",
    "specialinterest": "Interés especial",
    "spokenword": "Hablado",
    "sports": "Deportes",
    "sports&outdoors": "Deportes y vida al aire libre",
    "sports&recreation": "Deportes y ocio",
    "teaching&learning": "Educación y aprendizaje",
    "technology": "Tecnología",
    "teens": "Adolescentes",
    "thriller": "Thriller",
    "tokusatsu": "Tokusatsu",
    "topalbums": "Top de álbumes",
    "topaudiobooks": "Top de audiolibros",
    "topfreeapplications": "Top de apps gratuitas",
    "topfreeebooks": "Top de libros gratis",
    "topfreeipadapplications": "Top de apps gratis para iPad",
    "topfreemacapps": "Top de apps para Mac gratuitas",
    "topgrossingapplications": "Top de apps por ingresos",
    "topgrossingipadapplications": "Top de apps para iPad por ingresos",
    "topgrossingmacapps": "Top de apps para Mac por ingresos",
    "topimixes": "Top de listas iMix",
    "topitunesucollections": "Top de colecciones de iTunes U",
    "topitunesucourses": "Top de cursos de iTunesU",
    "topmacapps": "Top de apps para Mac",
    "topmovies": "Top de películas",
    "topmusicvideos": "Top de vídeos musicales",
    "toppaidapplications": "Top de apps de pago",
    "toppaidebooks": "Top de libros de pago",
    "toppaidipadapplications": "Top de apps de pago para iPad",
    "toppaidmacapps": "Top de apps para Mac de pago",
    "toppodcasts": "Top de podcasts",
    "topsongs": "Top de canciones",
    "toptextbooks": "Top Text Books",
    "topvideorentals": "Top de alquileres",
    "travel": "Viajes",
    "travel&adventure": "Viajes y aventura",
    "turkish": "Cine turco",
    "tv&film": "Cine y televisión",
    "urban": "Urbano",
    "utilities": "Utilidades",
    "video": "Vídeo",
    "vocal": "Vocal",
    "weather": "El tiempo",
    "western": "Western",
    "world": "Músicas del mundo",
  },
  "error": null,
  "isFetchingCommon": false,
  "isFetchingMediaType": false,
}
`;

exports[`translations reducer should return the initial state 1`] = `
Object {
  "data": Object {},
  "error": null,
  "isFetchingCommon": false,
  "isFetchingMediaType": false,
}
`;
